drivers/modem/gsm_ppp.c

```diff
@@ -1267,24 +1265,24 @@ static int gsm_init(const struct device *dev)
 	(void)k_mutex_init(&gsm->lock);
 	gsm->dev = dev;
 
-	gsm->cmd_handler_data.cmds[CMD_RESP] = response_cmds;
-	gsm->cmd_handler_data.cmds_len[CMD_RESP] = ARRAY_SIZE(response_cmds);
-	gsm->cmd_handler_data.match_buf = &gsm->cmd_match_buf[0];
-	gsm->cmd_handler_data.match_buf_len = sizeof(gsm->cmd_match_buf);
-	gsm->cmd_handler_data.buf_pool = &gsm_recv_pool;
-	gsm->cmd_handler_data.alloc_timeout = K_NO_WAIT;
-	gsm->cmd_handler_data.eol = "\r";
-
 	(void)k_sem_init(&gsm->sem_response, 0, 1);
 	(void)k_sem_init(&gsm->sem_if_down, 0, 1);
 
-	ret = modem_cmd_handler_init(&gsm->context.cmd_handler,
-				   &gsm->cmd_handler_data);
+	ret = modem_cmd_handler_init(&gsm->context.cmd_handler, &gsm->cmd_handler_data,
+		&gsm->cmd_match_buf[0], sizeof(gsm->cmd_match_buf), &gsm_recv_pool,
+		K_NO_WAIT, "\r", NULL);
```

Comment:
I don't think calling `modem_cmd_handler_init()` is readable enough now. IMO there are way too many arguments passed to this function. This makes it also error prone, as it is easy to pass arguments in wrong order, or pass something unrelevant which is just the same type.

What about defining a structure (possibly const to save RAM space), that would be initialized using "designated initializer", similar how work queue is initialized with help of `struct k_work_queue_config`?

Reply:
It's interesting that this approach is suggested considering that's how it was done previously. I think the reasoning for moving them to function arguments was to facilitate runtime checking of the parameters before they were written into the context structures.